roqoqo-iqm/src/interface.rs

```diff
@@ -62,51 +64,51 @@ pub struct IqmInstruction {
 // HashMap that associates to each register name the indices in the register that are being affected
 // by measurements. These indices are saved in the order in which the measurement operations appear
 // in the circuit, since this is the order in which the backend returns the results.
-pub(crate) type RegisterMapping = HashMap<String, Vec<usize>>;
+pub(crate) type MeasuredQubitsMap = HashMap<String, (Vec<usize>, usize)>;
 
-/// Converts all operations in a [roqoqo::Circuit] into instructions for IQM Hardware or IQM Simulators
+/// Converts all operations in a [roqoqo::Circuit] into instructions for IQM Hardware.
 ///
 /// # Arguments
 ///
 /// * `circuit` - The [roqoqo::Circuit] that is converted
-/// * `device_number_qubits` - The number of qubits of the backend device. It is used to know how many qubits to measure with [roqoqo::operations::PragmaRepeatedMeasurement]
-/// * `output_registers` - A mutable reference to the classical registers that need to be initialized
+/// * `device_number_qubits` - The number of qubits of the backend device. It is used to know how
+///    many qubits to measure with [roqoqo::operations::PragmaRepeatedMeasurement]
 /// * `number_measurements_internal` - If set, the number of measurements that has been overwritten
 /// in the backend
+/// * `circuit_index` - Index of the circuit in the batch, needed to assign a unique name to the circuit.
 ///
 /// # Returns
 ///
-/// * `Ok(IqmCircuit, RegisterMapping, usize)` - Converted circuit, mapping of measured qubits to register indices, and number of measurements
-/// * `Err(RoqoqoBackendError::OperationNotInBackend)` - Error when [roqoqo::operations::Operation] can not be converted
+/// * `Ok(IqmCircuit, usize)` - Converted circuit and number of measurements
+/// * `Err(RoqoqoBackendError::OperationNotInBackend)` - Error when [roqoqo::operations::Operation]
+///    can not be converted
 pub fn call_circuit<'a>(
     circuit: impl Iterator<Item = &'a Operation>,
     device_number_qubits: usize,
-    output_registers: &mut HashMap<String, BitOutputRegister>,
     number_measurements_internal: Option<usize>,
-) -> Result<(IqmCircuit, RegisterMapping, usize), RoqoqoBackendError> {
+    circuit_index: usize,
+) -> Result<(IqmCircuit, usize), IqmBackendError> {
     let mut circuit_vec: Vec<IqmInstruction> = Vec::new();
     let mut number_measurements: usize = 1;
     let mut measured_qubits: Vec<usize> = vec![];
-    let mut register_mapping: RegisterMapping = HashMap::new();
+    let mut measured_qubits_map: MeasuredQubitsMap = HashMap::new();
 
     for op in circuit {
         match op {
             Operation::DefinitionBit(o) => {
-                // initialize output registers with default `false` values
```

Comment:
Why have we moved from using false everywhere to nothing?

Reply:
I moved from initializing the output registers when submitting the circuits, to doing it when we post-process the results, to allow for asynchronous execution.